plugin/cmd/kube-scheduler/app/server.go

```diff
@@ -110,9 +111,41 @@ func Run(s *options.SchedulerServer) error {
 	eventBroadcaster.StartRecordingToSink(kubeClient.Events(""))
 
 	sched := scheduler.New(config)
-	sched.Run()
 
-	select {}
+	run := func(_ <-chan struct{}) {
+		sched.Run()
+		select {}
+	}
+
+	if !s.LeaderElection.LeaderElect {
+		run(nil)
```

Comment:
I must be missing something -- while can't you just make this block say
```
sched.Run()
select {}
```
instead of having a separate run() function? Is run() used from somewhere else that I missed?